cpp/src/reductions/collect_ops.cu

```diff
@@ -0,0 +1,96 @@
+                                        nan_equality nans_equal,
+                                        rmm::cuda_stream_view stream,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  auto list_wrapper   = lists::detail::make_lists_column_from_scalar(scalar, 1, stream, mr);
+  auto lcw            = lists_column_view(list_wrapper->view());
+  auto no_dup_wrapper = lists::drop_list_duplicates(lcw, nulls_equal, nans_equal, mr);
+  auto no_dup         = lists_column_view(no_dup_wrapper->view()).get_sliced_child(stream);
+  return make_list_scalar(no_dup, stream, mr);
+}
+
+std::unique_ptr<scalar> collect_list(column_view const& col,
+                                     null_policy null_handling,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr)
+{
+  if (null_handling == null_policy::EXCLUDE && col.has_nulls()) {
+    auto d_view             = column_device_view::create(col, stream);
+    auto filter             = detail::validity_accessor(*d_view);
+    auto null_purged_table  = detail::copy_if(table_view{{col}}, filter, stream, mr);
+    column* null_purged_col = null_purged_table->release().front().release();
+    null_purged_col->set_null_mask(rmm::device_buffer{0, stream, mr}, 0);
+    return std::make_unique<list_scalar>(std::move(*null_purged_col), true, stream, mr);
+  } else {
+    return make_list_scalar(col, stream, mr);
+  }
+}
+
+std::unique_ptr<scalar> merge_lists(column_view const& col,
+                                    rmm::cuda_stream_view stream,
+                                    rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(col.type().id() == type_id::LIST,
+               "input column of merge_lists must be a list column");
+  auto flatten_col = lists_column_view(col).get_sliced_child(stream);
+  return make_list_scalar(flatten_col, stream, mr);
+}
+
+std::unique_ptr<scalar> collect_set(column_view const& col,
+                                    null_policy null_handling,
+                                    null_equality nulls_equal,
+                                    nan_equality nans_equal,
+                                    rmm::cuda_stream_view stream,
+                                    rmm::mr::device_memory_resource* mr)
+{
+  auto scalar = collect_list(col, null_handling, stream, mr);
+  auto ls     = dynamic_cast<list_scalar*>(scalar.get());
+  return drop_duplicates(*ls, nulls_equal, nans_equal, stream, mr);
+}
+
+std::unique_ptr<scalar> merge_sets(column_view const& col,
+                                   null_equality nulls_equal,
+                                   nan_equality nans_equal,
+                                   rmm::cuda_stream_view stream,
+                                   rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(col.type().id() == type_id::LIST,
+               "input column of merge_lists must be a list column");
```

Comment:
It's more descriptive to just make the parameter a `lists_column_view` and eliminates the extra error checking. 
```suggestion
std::unique_ptr<scalar> merge_sets(lists_column_view const& col,
                                   null_equality nulls_equal,
                                   nan_equality nans_equal,
                                   rmm::cuda_stream_view stream,
                                   rmm::mr::device_memory_resource* mr)
{
```

Reply:
Done.